python/private/pypi/parse_requirements.bzl

```diff
@@ -46,6 +46,19 @@ def _extract_version(entry):
         return version
     return None
 
+def _extract_marker_string(entry):
+    """Extract the marker string from the requirement string.
+
+    Args:
+        entry: {type}`str` The requirement string.
+    """
+    marker_start = entry.find(";")
+    if marker_start != -1:
+        # Extract everything after ';' until --hashes if it exists or end of the string
+        marker, _, _ = entry[marker_start + 1:].partition(" --hashes")
+        return marker.strip()
+    return ""
```

Comment:
I have a `pep508_requirement.bzl` file that is very good for parsing these sort of things. We should probably use it?